collector/infiniband_linux.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+// +build !noinfiniband
+
+package collector
+
+import (
+	"errors"
+	"path/filepath"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+const infinibandPath = "class/infiniband"
+
+var (
+	errInfinibandNoDevicesFound = errors.New("no InfiniBand devices detected")
+	errInfinibandNoPortsFound   = errors.New("no InfiniBand ports detected")
+)
+
+type infinibandCollector struct {
+	metricDescs map[string]*prometheus.Desc
+	counters    map[string]string
+}
+
+func init() {
+	Factories["infiniband"] = NewInfiniBandCollector
+}
+
+func NewInfiniBandCollector() (Collector, error) {
+	var i infinibandCollector
+
+	// Filenames of all InfiniBand counter metrics including a detailed description.
+	i.counters = map[string]string{
+		"link_downed":            "Number of times the link failed to recover from an error state and went down",
+		"link_error_recovery":    "Number of times the link successfully recovered from an error state",
+		"multicast_rcv_packets":  "Number of multicast packets received (including errors)",
+		"multicast_xmit_packets": "Number of multicast packets transmitted (including errors)",
+		"port_rcv_data":          "Number of data octets received on all links",
```

Comment:
See discussion above: Using good names if we have to track them here anyway or us the filenames here. If we add bytes here, we should also change rcv to receive etc etc, right?
I lean towards making all names follow our best practices but fine either way.